manifests/api.pp

```diff
@@ -16,6 +16,10 @@
 class razor::api (
   String $hostname = 'localhost',
   Variant[Undef, Integer] $port = undef,
+  Variant[String, Enum['http', 'https']] $http_method = 'http',
```

Comment:
@Nekototori Should this parameter be only one of `http` and `https`?

If so, you don't need to Variant String stuff, and can do just `Enum['http', 'https']`.